### ISSUE TITLE
Calls on mixed value are not flagged

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1313]
https://getpsalm.org/r/bfe3b4690f
```php
<?php
  /** 
   * @var mixed $s
   * @psalm-suppress MixedAssignment
   */
  $s = 1;

  $s(); // should be flagged

  Closure::fromCallable($s); // is flagged
```
**Expected**: MixedFunctionCall + MixedArgument
**Actual**: only MixedArgument emitted
```
Psalm output (using commit 80c9114): 

INFO: MixedArgument - 10:25 - Argument 1 of fromCallable cannot be mixed, expecting callable
```
